msteams-platform/m365-apps/using-teams-client-sdk-preview.md

```diff
@@ -0,0 +1,365 @@
+---
+title: Microsoft Teams JavaScript client SDK v2 Preview
+description: Understand the changes coming with Microsoft Teams JavaScript client SDK v2 Preview
+ms.date: 11/15/2021
+ms.topic: conceptual
+---
+# Microsoft Teams JavaScript client SDK v2 Preview
+
+With the Microsoft Teams JavaScript client SDK v2 Preview, the [existing Teams SDK](/javascript/api/@microsoft/teams-js/) (`@microsoft/teams-js`, or simply `TeamsJS`) has been refactored to enable Teams developers the ability to [extend Teams apps to run in Outlook and Office](overview.md). From a functional perspective, the TeamsJS SDK v2 Preview (`@microsoft/teams-js@2.0.0-beta`) is a superset of the current TeamsJS SDK: it supports existing Teams app functionality while adding the ability to host Teams apps in Outlook and Office.
+
+There are two significant changes in the TeamsJS SDK v2 Preview that your code will need to account for in order to run in other Microsoft 365 applications:
+
+* [**Callback functions now return Promise objects.**](#callbacks-converted-to-promises) All existing functions with a callback parameter have been modernized to return a JavaScript [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) object for improved handling of asynchronous operations and code readability.
+
+ - [**APIs are now organized into *capabilities*.**](#apis-organized-into-capabilities) You can think of capabilities as logical groupings of APIs that provide similar functionality, such as `authentication`, `calendar`, `mail`, `monetization`, `meeting`, and `sharing`.
+ 
+ You can use the [Teams Toolkit extension](https://aka.ms/teams-toolkit) for Visual Studio Code to simplify the update process for your Teams app, as described in the next section.
+
+> [!NOTE]
+> Enabling an existing Teams app to run in Outlook and Office requires both:
+> 1. Dependency on the `@microsoft/teams-js@2.0.0-beta.0` or higher, and
+> 2. Modifying existing application code according to the required changes described in this document.
+>
+>  If you reference the `@microsoft/teams-js@2.0.0-beta.0` SDK version from an existing Teams app, you will see deprecation warnings if your code calls APIs that have changed. An API translation layer (mapping current SDK to preview SDK API calls) is provided to enable existing Teams apps to continue working in Teams until they are able to update code to work with the TeamsJS SDK v2 Preview. Once you update your code with the changes outlined in this article, your personal tab will also run in Outlook and Office.
```

Comment:
We should either update the version to be `@microsoft/teams-js@2.0.0-beta.1` or use `@microsoft/teams-js@next` since  we have deprecated 2.0.0-beta.0.